include/deal.II/lac/block_linear_operator.h

```diff
@@ -26,23 +26,185 @@
 DEAL_II_NAMESPACE_OPEN
 
 // Forward declarations:
+
 template <typename Number> class BlockVector;
 
+template <size_t m,
+          size_t n,
+          typename Range = BlockVector<double>,
+          typename Domain = Range>
+class BlockLinearOperator;
+
+template <size_t m, size_t n,
+          typename Range = BlockVector<double>,
+          typename Domain = Range,
+          typename BlockMatrix>
+BlockLinearOperator<m, n, Range, Domain>
+block_operator(const BlockMatrix &matrix);
+
+template <size_t m, size_t n,
+          typename Range = BlockVector<double>,
+          typename Domain = Range>
+BlockLinearOperator<m, n, Range, Domain>
+block_operator(const std::array<std::array<LinearOperator<typename Range::BlockType, typename Domain::BlockType>, n>, m> &);
+
+// This is a workaround for a bug in <=gcc-4.7 that does not like partial
+// template default values in combination with local lambda expressions [1]
+//
+// [1] https://gcc.gnu.org/bugzilla/show_bug.cgi?id=53624
+//
+// Forward declare functions with partial template defaults:
```

Comment:
This is going to be picked up by doxygen. You could of course use the opportunity to just put the documentation here. If you'd like to put the documentation elsewhere, you may want to consider enclosing all of this by `#ifndef DOXYGEN ... #endif`